### ISSUE TITLE
quotemark config option ordering matters?

### ISSUE THREAD

**Issue by alexwalter88** [palantir/tslint#3063]
### Bug Report

- __TSLint version__: 5.5.0
- __TypeScript version__: 2.4.2
- __Running TSLint via__: tslint-language-service in VSCode

#### TypeScript code being linted

```ts
let k = "blah";
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "quotemark": [
      true,
      "avoid-template",
      "single"
    ]
  }
}
```

#### Actual behavior
no error
#### Expected behavior
`" should be ' (quotemark)` error

If I put `"single"` after `true` in the config array, then it works. I tested rearranging the other possible config options like `"jsx-double"`, and those all seem to work regardless of order, but as soon as `"single"` is not the first config option, the entire rule stops working.

Is this by design or a bug? I couldn't see any documentation on the ordering of config options mattering. Could this be affecting other rules too? I ordered all of my config options alphabetically because I assumed order wouldn't matter.

**Comment by alexwalter88** [comment]
Welp found the place in the code is explicitly checking the first value of the config options: https://github.com/palantir/tslint/blob/master/src/rules/quotemarkRule.ts#L78-L85

I would have seen the error too if I was not using tslint-language-service, but Typescript doesn't yet support logging from plugins for end users. Although, this should probably be mentioned in the rule documentation.

**Comment by ajafff** [comment]
You're right, this rule expects `"single"` or `"double"` to be the first option.
That should at least be documented.

#889 also tracks validating the config to show an error in such cases.

**Comment by aervin** [comment]
Would it be better if we searched through argument list to find OPTION_SINGLE or OPTION_DOUBLE and throw error if no such argument provided?
